### PR TITLE
Update `@nextcloud/files` to 3.6.0 and fix display name handling of folders (breadcrumbs and filename)

### LINKED ISSUE #46604: [Bug]: new name of renamed file only shown after page reload
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

..

### Steps to reproduce

1. Rename a file on c.nc.com 
2. Infobox says it is successfully renamed, but the old name is still shown in the files list. The new name is only shown after page reload. 

### Expected behavior

New filename shows instantly.

### Installation method

None

### Nextcloud Server version

29

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

29.0.4 RC1

### LINKED ISSUE #46660: [Bug]: Renamed files are not refreshed in the web interface + side weird issues
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When a file is renamed, old file name remains unless page is maually refresh or browsed aways.

There are alow ohter side weird issues; if selecting renamed file, we have the option to remove it from favourites, while file isn't there (!!) and also if deleting it, it reamains on the interface (but is deleted). 🤒 

### Steps to reproduce

1. Right click on a file
2. Rename it
3. See old file name

image

### Expected behavior

New file name is visible. Reproducible in 28 and 29.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

29

### Additional info

https://github.com/user-attachments/assets/1adfa50b-27a7-457b-ad28-8ee07a7c646d

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/server/issues/46660
* Resolves https://github.com/nextcloud/server/issues/46604
* Resolves parts of https://github.com/nextcloud/groupfolders/issues/3068#issuecomment-2242742173
* Closes https://github.com/nextcloud/server/pull/46474

## Summary

1. Update to current version 3.6.0 of nextcloud-files (drop duplicated code)
2. For folders do not cut-off the extension as folders do not have extensions they are part of the filename, this fixes names of folder in the file list and in the bread crumbs

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/components/FileEntryMixin.ts [hunk 1]
```diff
@@ -74,19 +74,31 @@ export default defineComponent({
...
+		 * The display name of the current node
+		 * Either the nodes filename or a custom display name (e.g. for shares)
+		 */
+		displayName() {
+			return this.source.displayname
+		},
+		/**
+		 * The display name without extension
+		 */
+		basename() {
```
Comment (artonge) [inline apps/files/src/components/FileEntryMixin.ts:87]: Is basename the correct word? If I use `basename` in my terminal I have:

```shell
$ basename README.md 
README.md
```

Does it really make sense to have the filename without its extension?

Comment (artonge) [reply apps/files/src/components/FileEntryMixin.ts:87]: Hmm, anyway, the comment is explicit enough :)

**Review by artonge — APPROVED** [review]

**Review by nfebe — APPROVED** [review]

**Pull request merged**